### PR TITLE
Added a channel_id_from_name method on Guild. Fixing #571

### PR DESCRIPTION
**Pull request by xacrimon**

### CODE REVIEW

**Comment by xacrimon** [comment]
Fix #571

File: src/model/guild/mod.rs [hunk 1]
```diff
@@ -206,6 +206,20 @@ impl Guild {
         permissions.is_empty()
     }
 
+    #[cfg(feature = "cache")]
+    pub fn channel_id_from_name(&self, cache: impl AsRef<CacheRwLock>, name: impl AsRef<str>) -> Option<ChannelId> {
+        let name = name.as_ref();
+        let cache = cache.as_ref().read();
+        let guild = cache.guilds.get(&self.id)?.read();
+
+        guild.channels
```
Comment (arqunis) [inline src/model/guild/mod.rs:215]:
```rust, suggestion
guild.channels
    .iter()
    .find_map(|(id, c)| {
        if c.read().name == name { 
            Some(*id) 
        } else { 
            None
        }
     })
```

**Review by arqunis — CHANGES_REQUESTED** [review]

**Review by xacrimon — COMMENTED** [review]
Looks good

**Comment by xacrimon** [comment]
I'll update the branch when get home

**Comment by xacrimon** [comment]
@acdenisSK Changes have been made according to you request.

**Pull request merged**